src/RVRP_data.jl

```diff
@@ -20,57 +14,78 @@ struct TimeWindow
     end_time::Float64
 end
 
-struct UnitPricing
+struct UnitPrices # cost per unit
     travel_distance_price::Float64
     travel_time_price::Float64
     service_time_price::Float64
-    wait_time_price::Float64
+    waiting_time_price::Float64
 end
 
 mutable struct Depot
     id::String
     index::Int # Not given in JSON
     location::Location
-    time_windows::Vector{TimeWindow} # optional
+    opening_time_windows::Vector{TimeWindow} # optional
 end
 
-mutable struct Pickup
+mutable struct PickupPoint
     id::String # If its part of a shipment, it has is own id anyway
-    index::Int # Not given in JSON. If its part of a shipment, the index of the shipment
+    index::Int # Not given in JSON. 
     location::Location
-    shipment_id::String # NULL if none
-    price_reward::Float64
-    capacity_request::Float64
-    time_windows::Vector{TimeWindow} # optional
-    service_time::Float64 # optional
+    opening_time_windows::Vector{TimeWindow} # optional
+    service_time::Float64 # optional; to measure access time for instance
 end
 
-mutable struct Delivery
+
+mutable struct DeliveryPoint
     id::String # If it is part of a shipment,it has is own id anyway
-    index::Int # Not given in SON. if it is part of a shipment, the index of the shipment
+    index::Int # Not given in JSON. 
     location::Location
-    shipment_id::String # NULL if none
-    price_reward::Float64
-    capacity_request::Float64
-    time_windows::Vector{TimeWindow} # optional
-    service_time::Float64 # optional
+    opening_time_windows::Vector{TimeWindow} # optional
+    service_time::Float64 # optional; to measure access time for instance
+end
+
+mutable struct ProductCategory
+    id::String 
+    index::Int # Not given in JSON. 
+    conflicting_product_ids::Vector{String} # if any
+    prohibited_predecessor_product_ids::Vector{String}  # if any
 end
 
-mutable struct Shipment
+mutable struct SpecificProduct # an entity to understand as a commodity in a multi-commodity flow problem
+    id::String 
+    index::Int # Not given in JSON. 
+    product_category_id::String
+    stock_availabitilies_at_pickup_points::Dict{String,Float64} # undefined if no restrictions, i.e, if available in any PickupPoint in large quantities
+    stock_capacities_at_delivery_points::Dict{String,Float64} # undefined if no restrictions, i.e, if one can deliver to any DeliveryPoint in large quantities
+end
+
+mutable struct ShipmentRequest
     id::String
     index::Int # Not given in JSON
-    price_reward::Float64
-    pickup::Pickup
-    delivery::Delivery
-    max_duration::Float64
+    request_type::Int # 1 = pickup, 2 = delivery, 3 = pickup&delivery, 4 = complex_request
```

Comment:
I m not sure I like the request_type especially when you use it after to give slightly different meaning to attriubutes...